src/DAO/scheduleDAO.ts

```diff
@@ -21,6 +35,42 @@ export class ScheduleDAO
     return await scheduleRepository.find({ relations: ['trainer', 'class'] });
   }
 
+  private prepareSort(options: ISearchParams): OrderByCondition {
+    const orderParams: OrderByCondition = {};
+    if (options.sort && options.sortBy) {
+      orderParams[options.sortBy] = options.sort === 'ASC' ? 'ASC' : 'DESC';
+    }
+    return orderParams;
+  }
+  private prepareFilters(
+    options: ISearchParams,
+  ): { trainer: User; startDate: FindOperator<Date> } {
+    const searchParams = {
+      trainer: options.trainer,
+      startDate: Between(options.fromDate, options.untilDate),
+    };
+    if (!options.trainer) {
```

Comment:
these "if's" could be simplified:


```js
const searchParams: ISearchParams = {};

Object.keys(options).map(key => {  
  if (searchParams[key]) {
    obj[key] = filtersToApply[key];
  } 
});

```

something along those lines (not sure it's working code)